### ISSUE TITLE
Using the docker service, no response is gotten when using the console web app

### ISSUE THREAD

**Issue by xdSora** [kermitt2/datastet#7]
After following the instructions in the README and running:
```
docker run --rm --gpus all -it --init --ulimit core=0 -p 8060:8060 grobid/datastet:0.8.0
```

I was able to access the console on `localhost:8060` in Firefox. When attempting to use the Datastet service "Process text sentence" and using one of the example filler texts, this is the output of the running docker service after several minutes (the request never finishes and no response is returned):

```
init classifier...
allenai/scibert_scivocab_cased will be used, loaded via delft_model
---
max_epoch: 50
early_stop: True
batch_size (training): 16
maxlen: 300
model_name: dataseer-binary_bert
learning_rate:  0.001
---
Model: "model"
_________________________________________________________________
 Layer (type)                Output Shape              Param #
=================================================================
 input_token (InputLayer)    [(None, None)]            0

 tf_bert_model (TFBertModel)  TFBaseModelOutputWithPoo  109938432
                             lingAndCrossAttentions(l
                             ast_hidden_state=(None,
                             None, 768),
                              pooler_output=(None, 76
                             8),
                              past_key_values=None, h
                             idden_states=None, atten
                             tions=None, cross_attent
                             ions=None)

 dropout_37 (Dropout)        (None, 768)               0

 dense (Dense)               (None, 2)                 1538

=================================================================
Total params: 109,939,970
Trainable params: 109,939,970
Non-trainable params: 0
_________________________________________________________________
load weights from /opt/grobid/datastet/../grobid-home/models/dataseer-binary_bert/model_weights.hdf5
loading model weights /opt/grobid/datastet/../grobid-home/models/dataseer-binary_bert/model_weights.hdf5
init classifier...
allenai/scibert_scivocab_cased will be used, loaded via delft_model
---
max_epoch: 50
early_stop: True
batch_size (training): 16
maxlen: 300
model_name: dataseer-first_bert
learning_rate:  0.001
---
Model: "model_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #
=================================================================
 input_token (InputLayer)    [(None, None)]            0

 tf_bert_model_1 (TFBertMode  TFBaseModelOutputWithPoo  109938432
 l)                          lingAndCrossAttentions(l
                             ast_hidden_state=(None,
                             None, 768),
                              pooler_output=(None, 76
                             8),
                              past_key_values=None, h
                             idden_states=None, atten
                             tions=None, cross_attent
                             ions=None)

 dropout_75 (Dropout)        (None, 768)               0

 dense_1 (Dense)             (None, 25)                19225

=================================================================
Total params: 109,957,657
Trainable params: 109,957,657
Non-trainable params: 0
_________________________________________________________________
load weights from /opt/grobid/datastet/../grobid-home/models/dataseer-first_bert/model_weights.hdf5
loading model weights /opt/grobid/datastet/../grobid-home/models/dataseer-first_bert/model_weights.hdf5
init classifier...
allenai/scibert_scivocab_cased will be used, loaded via delft_model
---
max_epoch: 50
early_stop: True
batch_size (training): 16
maxlen: 300
model_name: dataseer-reuse_bert
learning_rate:  0.001
---
Model: "model_2"
_________________________________________________________________
 Layer (type)                Output Shape              Param #
=================================================================
 input_token (InputLayer)    [(None, None)]            0

 tf_bert_model_2 (TFBertMode  TFBaseModelOutputWithPoo  109938432
 l)                          lingAndCrossAttentions(l
                             ast_hidden_state=(None,
                             None, 768),
                              pooler_output=(None, 76
                             8),
                              past_key_values=None, h
                             idden_states=None, atten
                             tions=None, cross_attent
                             ions=None)

 dropout_113 (Dropout)       (None, 768)               0

 dense_2 (Dense)             (None, 2)                 1538

=================================================================
Total params: 109,939,970
Trainable params: 109,939,970
Non-trainable params: 0
_________________________________________________________________
load weights from /opt/grobid/datastet/../grobid-home/models/dataseer-reuse_bert/model_weights.hdf5
loading model weights /opt/grobid/datastet/../grobid-home/models/dataseer-reuse_bert/model_weights.hdf5
BERT_CRF
michiyasunaga/LinkBERT-base will be used, loaded via delft_model
load weights from /opt/grobid/datastet/../grobid-home/models/datasets-BERT_CRF/model_weights.hdf5
loading model weights /opt/grobid/datastet/../grobid-home/models/datasets-BERT_CRF/model_weights.hdf5
Model: "model"
__________________________________________________________________________________________________
 Layer (type)                   Output Shape         Param #     Connected to
==================================================================================================
 input_token (InputLayer)       [(None, None)]       0           []

 input_attention_mask (InputLay  [(None, None)]      0           []
 er)

 input_token_type (InputLayer)  [(None, None)]       0           []

 tf_bert_model (TFBertModel)    TFBaseModelOutputWi  108310272   ['input_token[0][0]',
                                thPoolingAndCrossAt               'input_attention_mask[0][0]',
                                tentions(last_hidde               'input_token_type[0][0]']
                                n_state=(None, None
                                , 768),
                                 pooler_output=(Non
                                e, 768),
                                 past_key_values=No
                                ne, hidden_states=N
                                one, attentions=Non
                                e, cross_attentions
                                =None)

 dropout_37 (Dropout)           (None, None, 768)    0           ['tf_bert_model[0][0]']

==================================================================================================
Total params: 108,310,272
Trainable params: 108,310,272
Non-trainable params: 0
__________________________________________________________________________________________________
Model: "crf_model_wrapper_for_bert"
_________________________________________________________________
 Layer (type)                Output Shape              Param #
=================================================================
 crf (CRF)                   multiple                  6232

 model (Functional)          (None, None, 768)         108310272

=================================================================
Total params: 108,316,504
Trainable params: 108,316,504
Non-trainable params: 0
_________________________________________________________________
```
I have tried reinstalling it several times, and have waited up to 15 minutes for the request to finish, with no success. The [GROBID](https://github.com/kermitt2/grobid) docker service works as expected.

The docker image also seems to be very unresponsive. When stopping it via CTRL-C, it takes ~20 seconds to actually stop, and returns this additional output before doing so:
```
^CWARN  [2024-10-03 10:05:43,391] org.eclipse.jetty.util.thread.QueuedThreadPool: InstrumentedQueuedThreadPool[dw]@353c6da1{STOPPING,8<=6<=2048,i=0,r=-1,q=8}[NO_TRY] Couldn't stop Thread[dw-63 - POST /service/annotateDatasetPDF,5,main]
WARN  [2024-10-03 10:05:43,392] org.eclipse.jetty.util.thread.QueuedThreadPool: InstrumentedQueuedThreadPool[dw]@353c6da1{STOPPING,8<=6<=2048,i=0,r=-1,q=8}[NO_TRY] Couldn't stop Thread[dw-60 - GET /service/annotateDatasetSentence?text=We+found+that+this+technique+works+very+well+in+practice%2C+for+the+MNIST+and+NORB+datasets+(see+below).,5,main]
```
### Docker 
```
Client:
 Version:           27.2.0
 API version:       1.47
 Go version:        go1.23.0
 Git commit:        3ab4256958
 Built:             Thu Aug 29 16:44:26 2024
 OS/Arch:           linux/amd64
 Context:           default

Server:
 Engine:
  Version:          27.2.0
  API version:      1.47 (minimum version 1.24)
  Go version:       go1.23.0
  Git commit:       3ab5c7d003
  Built:            Thu Aug 29 16:44:26 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.7.22
  GitCommit:        7f7fdf5fed64eb6a7caf99b3e12efcf9d60e311c.m
 runc:
  Version:          1.1.14
  GitCommit:
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```
### nvidia-container-toolkit
`1.16.1-3`

### Nvidia drivers
`NVIDIA-SMI 550.120                Driver Version: 550.120        CUDA Version: 12.4`

**Comment by kermitt2** [comment]
Hello @xdSora !
What are the capacities of your server running the docker container? 
Running this service really require GPU, it is much more deep learning intensive than Grobid. Without GPU all the models (several fine-tuned SciBERT) will be loaded in memory and the service might start to swap with such unresponsive behavior. 
The service runs well with a GPU with 8GB VRAM or more, and 32 GB RAM.

**Comment by xdSora** [comment]
Hello! I tried running it on the following hardware:

CPU: AMD Ryzen 7 5800X (16) @ 3.800GHz
GPU: NVIDIA GeForce RTX 3080 12GB
Memory: 32GB DDR4

I was under the impression this would be enough to run it, and it seems highly unlikely that processing a single sentence would take such a long time.

**Comment by kermitt2** [comment]
Yes indeed, it should work perfectly well on your hardware. 

I reproduced the problem on a similar machine, unresponsive for a few minutes, but for me 6-7 minutes after the first query everything worked fine as expected. One sentence is then processed in around 100ms. A PDF (20 pages) is processed in 5s. 

Maybe sending a curl request first rather than a request with the web console would improve the situation for you? 

> curl -X POST -d "text=Insulin levels of all samples were measured by ELISA kit (Mercodia)." localhost:8060/service/annotateDatasetSentence

A warm-up for the first request is normal, because there is no preloading of the models at the start of the container, but it should be something like maximum 30s. I think it was working like that a few months ago after I built this docker image. I don't know why we have these 6-7 minutes unavailability now. Possibly a resource is missing in the docker image (a model or embeddings) and it is automatically downloaded, which could explain that nothing happen during several minutes.

I also rebuilt the docker image, upgraded to latest Grobid version, but the new image shows the same problem at startup. 

So this is for further investigate sorry... for the moment the only solution is to wait X minutes warm up :(

**Comment by kermitt2** [comment]
Problem was that the public disambiguation service (entity-fishing) uses a different url since a few months, so just needed an update in the config file in the docker image. As the service was not found during health check, there was a long long timeout...

It is fixed in the updated docker image `grobid/datastet:0.8.1`

**Comment by xdSora** [comment]
> Problem was that the public disambiguation service (entity-fishing) uses a different url since a few months, so just needed an update in the config file in the docker image. As the service was not found during health check, there was a long long timeout...
> 
> It is fixed in the updated docker image `grobid/datastet:0.8.1`

Hello, thank you for updating the docker image! I have tested the new version and it worked instantly :)
